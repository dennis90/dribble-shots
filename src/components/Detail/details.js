import React from 'react'

const Detail = ({shot}) => {
  return (
    <div className="shot-data">
      <div className="image">
        <img src={shot.images.hidpi} alt={shot.title} />
      </div>
      <div className="info">
        <div className="views-count">
          <span>{shot.views_count} views</span>
        </div>
        <div className="likes-count">
          <span>{shot.likes_count} likes</span>
        </div>
      </div>
    </div>
  )
}

export default Detail
