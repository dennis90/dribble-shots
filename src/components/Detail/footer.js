import React from 'react'


const Footer = ({shot}) => {
  return (
    <div dangerouslySetInnerHTML={{ __html: shot.description}}></div>
  )
}

export default Footer
