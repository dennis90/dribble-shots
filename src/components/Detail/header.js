import React from 'react'

const Header = ({shot}) => {
  return (
    <div className="shot-header">
      <div className="user-profile-image">
        <img src={shot.user.avatar_url} />
      </div>
      <div>
        <div className="shot-title">
          {shot.title}
        </div>
        <div className="user-profile-info">
          by <a href={shot.user.html_url} target="_blank">{shot.user.name}</a>
        </div>
      </div>
    </div>
  )
}

export default Header
