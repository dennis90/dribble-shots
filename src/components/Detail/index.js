import React from 'react'

import Header from './header'
import Details from './details'
import Footer from './footer'

import '../../assets/css/detail.css'

const Detail = ({shot}) => {
  if (shot) {
    return (
      <div className="shot-detail">
        <Header shot={shot} />
        <Details shot={shot} />
        <Footer shot={shot} />
      </div>
    )
  }
  return null
}

export default Detail
