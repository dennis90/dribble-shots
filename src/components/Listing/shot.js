import React from 'react'

import ShotFooter from './shot-footer'


const Shot = ({shot}) => {
  const getStyle = () => ({
      backgroundImage: `url(${shot.images.normal})`
  })

  return (
    <div className="shot" style={getStyle()}>
      <ShotFooter shot={shot} />
    </div>
  )
}

export default Shot
