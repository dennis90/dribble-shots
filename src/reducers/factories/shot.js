import {Factory} from 'rosie'

export const ShotFactory = new Factory()
  .sequence('id')
  .sequence('title', index => `Shot #${index}`)
  .sequence('description', index => `Shot Description #${index}`)
  .attrs({
    images: {
      hidpi: 'https://dribbble/images/hidpi',
      normal: 'https://dribbble/images/normal',
    },
    views_count: 100,
    likes_count: 100,
  })
  .attr('user', ['user'], user => UserFactory.build(user))

export const UserFactory = new Factory()
  .sequence('name', index => `User #${index}`)
  .attrs({
    avatar_url: 'https://dribbble.com/user/avatar.png',
    html_url: 'https://dribbble.com/user/',
  })
