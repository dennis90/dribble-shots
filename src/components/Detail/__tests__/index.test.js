import React from 'react'
import {shallow} from 'enzyme'

import Detail from '../index'
import {ShotFactory} from '../../../reducers/factories/shot'

describe('ShotDetail Component', () => {
  it('should render Header, Details and Footer Component', () => {
    const props = ShotFactory.build({})
    const wrapper = shallow(<Detail shot={props} />)
    expect(wrapper).toMatchSnapshot('Component mounting')
  })

  it('should return null if no `shot` is provided', () => {
    const wrapper = shallow(<Detail />)
    expect(wrapper).toMatchSnapshot('Component null mounting')
  })
})
