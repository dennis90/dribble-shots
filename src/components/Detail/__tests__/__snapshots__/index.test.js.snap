// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component mounting 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Detail
    shot={
        Object {
            "description": "Shot Description #1",
            "id": 1,
            "images": Object {
              "hidpi": "https://dribbble/images/hidpi",
              "normal": "https://dribbble/images/normal",
            },
            "likes_count": 100,
            "title": "Shot #1",
            "user": Object {
              "avatar_url": "https://dribbble.com/user/avatar.png",
              "html_url": "https://dribbble.com/user/",
              "name": "User #1",
            },
            "views_count": 100,
          }
    }
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": null,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <Header
          shot={
                    Object {
                              "description": "Shot Description #1",
                              "id": 1,
                              "images": Object {
                                "hidpi": "https://dribbble/images/hidpi",
                                "normal": "https://dribbble/images/normal",
                              },
                              "likes_count": 100,
                              "title": "Shot #1",
                              "user": Object {
                                "avatar_url": "https://dribbble.com/user/avatar.png",
                                "html_url": "https://dribbble.com/user/",
                                "name": "User #1",
                              },
                              "views_count": 100,
                            }
          }
/>,
        <Detail
          shot={
                    Object {
                              "description": "Shot Description #1",
                              "id": 1,
                              "images": Object {
                                "hidpi": "https://dribbble/images/hidpi",
                                "normal": "https://dribbble/images/normal",
                              },
                              "likes_count": 100,
                              "title": "Shot #1",
                              "user": Object {
                                "avatar_url": "https://dribbble.com/user/avatar.png",
                                "html_url": "https://dribbble.com/user/",
                                "name": "User #1",
                              },
                              "views_count": 100,
                            }
          }
/>,
        <Footer
          shot={
                    Object {
                              "description": "Shot Description #1",
                              "id": 1,
                              "images": Object {
                                "hidpi": "https://dribbble/images/hidpi",
                                "normal": "https://dribbble/images/normal",
                              },
                              "likes_count": 100,
                              "title": "Shot #1",
                              "user": Object {
                                "avatar_url": "https://dribbble.com/user/avatar.png",
                                "html_url": "https://dribbble.com/user/",
                                "name": "User #1",
                              },
                              "views_count": 100,
                            }
          }
/>,
      ],
      "className": "shot-detail",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": null,
        "nodeType": "function",
        "props": Object {
          "shot": Object {
            "description": "Shot Description #1",
            "id": 1,
            "images": Object {
              "hidpi": "https://dribbble/images/hidpi",
              "normal": "https://dribbble/images/normal",
            },
            "likes_count": 100,
            "title": "Shot #1",
            "user": Object {
              "avatar_url": "https://dribbble.com/user/avatar.png",
              "html_url": "https://dribbble.com/user/",
              "name": "User #1",
            },
            "views_count": 100,
          },
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": null,
        "nodeType": "function",
        "props": Object {
          "shot": Object {
            "description": "Shot Description #1",
            "id": 1,
            "images": Object {
              "hidpi": "https://dribbble/images/hidpi",
              "normal": "https://dribbble/images/normal",
            },
            "likes_count": 100,
            "title": "Shot #1",
            "user": Object {
              "avatar_url": "https://dribbble.com/user/avatar.png",
              "html_url": "https://dribbble.com/user/",
              "name": "User #1",
            },
            "views_count": 100,
          },
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": null,
        "nodeType": "function",
        "props": Object {
          "shot": Object {
            "description": "Shot Description #1",
            "id": 1,
            "images": Object {
              "hidpi": "https://dribbble/images/hidpi",
              "normal": "https://dribbble/images/normal",
            },
            "likes_count": 100,
            "title": "Shot #1",
            "user": Object {
              "avatar_url": "https://dribbble.com/user/avatar.png",
              "html_url": "https://dribbble.com/user/",
              "name": "User #1",
            },
            "views_count": 100,
          },
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": null,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <Header
            shot={
                        Object {
                                    "description": "Shot Description #1",
                                    "id": 1,
                                    "images": Object {
                                      "hidpi": "https://dribbble/images/hidpi",
                                      "normal": "https://dribbble/images/normal",
                                    },
                                    "likes_count": 100,
                                    "title": "Shot #1",
                                    "user": Object {
                                      "avatar_url": "https://dribbble.com/user/avatar.png",
                                      "html_url": "https://dribbble.com/user/",
                                      "name": "User #1",
                                    },
                                    "views_count": 100,
                                  }
            }
/>,
          <Detail
            shot={
                        Object {
                                    "description": "Shot Description #1",
                                    "id": 1,
                                    "images": Object {
                                      "hidpi": "https://dribbble/images/hidpi",
                                      "normal": "https://dribbble/images/normal",
                                    },
                                    "likes_count": 100,
                                    "title": "Shot #1",
                                    "user": Object {
                                      "avatar_url": "https://dribbble.com/user/avatar.png",
                                      "html_url": "https://dribbble.com/user/",
                                      "name": "User #1",
                                    },
                                    "views_count": 100,
                                  }
            }
/>,
          <Footer
            shot={
                        Object {
                                    "description": "Shot Description #1",
                                    "id": 1,
                                    "images": Object {
                                      "hidpi": "https://dribbble/images/hidpi",
                                      "normal": "https://dribbble/images/normal",
                                    },
                                    "likes_count": 100,
                                    "title": "Shot #1",
                                    "user": Object {
                                      "avatar_url": "https://dribbble.com/user/avatar.png",
                                      "html_url": "https://dribbble.com/user/",
                                      "name": "User #1",
                                    },
                                    "views_count": 100,
                                  }
            }
/>,
        ],
        "className": "shot-detail",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": null,
          "nodeType": "function",
          "props": Object {
            "shot": Object {
              "description": "Shot Description #1",
              "id": 1,
              "images": Object {
                "hidpi": "https://dribbble/images/hidpi",
                "normal": "https://dribbble/images/normal",
              },
              "likes_count": 100,
              "title": "Shot #1",
              "user": Object {
                "avatar_url": "https://dribbble.com/user/avatar.png",
                "html_url": "https://dribbble.com/user/",
                "name": "User #1",
              },
              "views_count": 100,
            },
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": null,
          "nodeType": "function",
          "props": Object {
            "shot": Object {
              "description": "Shot Description #1",
              "id": 1,
              "images": Object {
                "hidpi": "https://dribbble/images/hidpi",
                "normal": "https://dribbble/images/normal",
              },
              "likes_count": 100,
              "title": "Shot #1",
              "user": Object {
                "avatar_url": "https://dribbble.com/user/avatar.png",
                "html_url": "https://dribbble.com/user/",
                "name": "User #1",
              },
              "views_count": 100,
            },
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": null,
          "nodeType": "function",
          "props": Object {
            "shot": Object {
              "description": "Shot Description #1",
              "id": 1,
              "images": Object {
                "hidpi": "https://dribbble/images/hidpi",
                "normal": "https://dribbble/images/normal",
              },
              "likes_count": 100,
              "title": "Shot #1",
              "user": Object {
                "avatar_url": "https://dribbble.com/user/avatar.png",
                "html_url": "https://dribbble.com/user/",
                "name": "User #1",
              },
              "views_count": 100,
            },
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;

exports[`Component null mounting 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Detail />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): null,
  Symbol(enzyme.__nodes__): Array [
    null,
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
